python/lsst/daf/butler/registry/queries/_sql_query_backend.py

```diff
@@ -121,6 +121,7 @@ def filter_dataset_collections(
         result: dict[DatasetType, list[CollectionRecord]] = {
             dataset_type: [] for dataset_type in dataset_types
         }
+        summaries = self._managers.datasets.fetch_summaries(collections, result.keys())
```

Comment:
I'd prefer to move this fetch a little higher up the call stack, so if someone queries for many dataset types in the same list of collections in a single `queryDatasets` call, we can do one fetch for all dataset types in those collections.  This pattern in particular is something I'd like to make sure we optimize:
```
for result_for_dataset_type in butler.registry.queryDatasets(..., collections=x).byParentDatasetType():
    if result_for_dataset_type.any(exact=False, execute=False):
        print(f"Found {result_for_dataset_type.parentDatasetType.name}")
```
If that's too messy right now, we should keep it in mind for the new query interface.

Reply:
Another possibility might be be to pass the summaries down to this method and just fetch them in `SqlRegistry` methods.

Reply:
OK, I'll look at this later today, maybe I can add it on this branch still.

Reply:
I think this deserves a separate ticket too, I looked at what it needs and in some context we'd have to resolve both collection wildcards and dataset wildcards before fetching summaries. I want to think about it a bit more, maybe we need to change `fetch_summaries` to accept wildcards instead of just lists.